### PR TITLE
Add lazy import of OTC and 0xjs docs

### PR DESCRIPTION
**Pull request by LogvinovLeon**
This PR:
* Makes some routes lazily required
* Splits the app in 3 chunks required on demand

### CODE REVIEW

File: ts/globals.d.ts [hunk 1]
```diff
@@ -23,3 +23,16 @@ declare module '*.json' {
...
+declare module 'scroll-to-element' {
+    export = scrollToElement;
+}
+
+// This will be defined by default in TS 2.4
+// Source: https://github.com/Microsoft/TypeScript/issues/12364
+interface System {
+  import<T>(module: string): Promise<T>;
+}
+
```
Comment (fabioberger) [inline ts/globals.d.ts]: Let's remove the space here, since these lines are related.

File: ts/index.tsx [hunk 2]
```diff
@@ -65,6 +65,27 @@ const muiTheme = getMuiTheme({
     },
 });
 
+const createLazyComponent = (componentName: string, lazyImport: () => Promise<any>) => {
+    return (props: any) => {
+        const componentPromise = utils.asyncMap(mod => mod[componentName], lazyImport());
```
Comment (LogvinovLeon) [inline ts/index.tsx]: The problem is that I don't need to loop. I need to map. I can introduce the tempopary array, but I think that `asyncMap` can be used in future.
There are libraries for that, but we care about bundle size and dependency costs.

File: ts/lazy_component.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+export class LazyComponent extends React.Component<LazyComponentProps, LazyComponentState> {
+    constructor(props: LazyComponentProps) {
+        super(props);
+        this.state = {
+            component: undefined,
+        };
+    }
+    public async componentWillMount() {
```
Comment (LogvinovLeon) [inline ts/lazy_component.tsx]: It's essentially the same. React fires and forgets this method.
Yeah, I should be more explicit about it
https://medium.com/front-end-hacking/async-await-with-react-lifecycle-methods-802e7760d802

File: ts/lazy_component.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        const component = await props.reactComponentPromise;
+        this.setState({
+            component,
+        });
+    }
+}
+
+/**
+ * [createLazyComponent description]
+ * @param  componentName - name of exported component
```
Comment (fabioberger) [inline ts/lazy_component.tsx]: Can we get rid of the `-` between param name and description and align them with tabs to remain consistent with how we format comments in 0x.js.

File: ts/lazy_component.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+ * @param  lazyImport    - lambda returning module promise
+ *                         we pass a lambda because we only want to require a module if it's used
+ * @example `const LazyOTC = createLazyComponent('OTC', () => System.import<any>('ts/containers/otc'));``
+ */
+export const createLazyComponent = (componentName: string, lazyImport: () => Promise<any>) => {
+    return (props: any) => {
+        const reactComponentPromise = new Promise<React.ComponentClass<any>>((resolve, reject) => {
```
Comment (fabioberger) [inline ts/lazy_component.tsx]: Is there a reason why we are using Promises rather then `async/await` here?

**Pull request merged**